app/controllers/claims/support/schools_controller.rb

```diff
@@ -6,4 +6,44 @@ def index
   def show
     @school = Claims::School.find(params.require(:id))
   end
+
+  def new
+    @school = Claims::School.new
+  end
+
+  def check
+    if school.valid? && !school.claims?
+      @school = school.decorate
+    else
+      school.errors.add(:urn, :taken) if school.claims?
```

Comment:
Should we use the model validations for this instead?

```ruby
class Claims::School < School
  validates :urn, uniqueness: { scope: [claims: true] }
end
```

Reply:
As mentioned below, this wouldn't necessarily work because we would be working with the `School` model. Just in case the record already exists as a `Placements::School`. 
You could use `becomes` https://apidock.com/rails/ActiveRecord/Base/becomes, but this seems long winded just to add an error.

Reply:
Hmm, makes sense in terms of the constraint. I would avoid adding model errors into controller code though 🤔 Maybe we can extract this into a model method `#validate_urn` maybe?